### PR TITLE
new attempt at declaration lookup

### PR DESCRIPTION
**Pull request by robertylewis**
A new attempt at #21 based on @gebner 's suggestion. I was a little skeptical at first, but this does indeed seem to be better. See it in action: 
https://robertylewis.com/mathlib_docs/find/manifold

### CODE REVIEW

**Comment by robertylewis** [comment]
This doesn't work with the suggested local testing using `python -m http.server` since it won't serve files without .html extensions with the right headers. GitHub Pages does this though.

File: print_docs.py [hunk 1]
```diff
@@ -646,6 +646,20 @@ def write_site_map(partition):
     out.write(site_root + filename + '.html\n')
   out.close()
 
+def redirect_body(decl_name, decl_loc):
+  url = filename_core(site_root, decl_loc, 'html')
+  return f"""
+<!DOCTYPE html>
+<Content-Type="text/html">
```
Comment (gebner) [inline print_docs.py:653]:
```suggestion
<meta http-equiv="content-type" content="text/html">
```
Are these two lines necessary at all?

Comment (robertylewis) [reply print_docs.py:653]: Probably not, I'll test. This was part of my failed attempt to make it work with the Python local server.

**Comment by gebner** [comment]
An idea to make `http.server` work: we could generate files like `find/nat.add/index.html` and then link to `find/nat.add`.

**Comment by robertylewis** [comment]
Ah, much better!
https://robertylewis.com/mathlib_docs/find/units.coe_dvd

**Comment by robertylewis** [comment]
While we're at it, maybe it's time to make CI force push to the `mathlib_docs` repo. There's no need to track history there. And this will add > 40k files to that repo. They won't change all that much, but still.

**Comment by robertylewis** [comment]
A new option: we can also redirect to location in the source code.
https://robertylewis.com/mathlib_docs/find/matrix.det/src

**Comment by robertylewis** [comment]
I'm done adding features; ready when @gebner and/or @bryangingechen are happy!

File: print_docs.py [hunk 2]
```diff
@@ -99,6 +99,27 @@ def library_link(filename, line=None):
...
+  return False
+    
+
+def library_link_from_decl_name(decl_name, decl_loc, file_map):
+  try:
+    e = next(d for d in file_map[decl_loc] if name_in_decl(decl_name, d))
+  except StopIteration:
+    if decl_name[-3:] == '.mk':
+      return library_link_from_decl_name(decl_name[:-3], decl_loc, file_map)
+    print(decl_name)
```
Comment (gebner) [inline print_docs.py:118]: This is debugging code, right?

Comment (robertylewis) [reply print_docs.py]: Indeed. I've left it in, but clarified the debug output.

**Pull request merged**